src/ucp/wireup/wireup.c

```diff
@@ -1758,6 +1713,60 @@ ucs_status_t ucp_wireup_connect_remote(ucp_ep_h ep, ucp_lane_index_t lane)
     return status;
 }
 
+static void
+ucp_wireup_fill_is_connected_params(uct_ep_is_connected_params_t *params,
+                                    const ucp_address_entry_t *addr_entry)
+{
+    params->field_mask = 0;
+    if (addr_entry->dev_addr != NULL) {
+        params->device_addr = addr_entry->dev_addr;
+        params->field_mask |= UCT_EP_IS_CONNECTED_FIELD_DEVICE_ADDR;
+    }
+
+    if (addr_entry->iface_addr != NULL) {
+        params->iface_addr  = addr_entry->iface_addr;
+        params->field_mask |= UCT_EP_IS_CONNECTED_FIELD_IFACE_ADDR;
+    }
+}
+
+static ucp_lane_index_t
+ucp_wireup_get_remote_lane(uct_ep_h uct_ep,
+                           const ucp_address_entry_t *addr_entry)
+{
+    uct_ep_is_connected_params_t params;
+    unsigned ep_index;
+
+    uct_ep = ucp_wireup_ep_get_tl_ep(uct_ep);
+    ucp_wireup_fill_is_connected_params(&params, addr_entry);
+    params.field_mask |= UCT_EP_IS_CONNECTED_FIELD_EP_ADDR;
+
+    for (ep_index = 0; ep_index < addr_entry->num_ep_addrs; ++ep_index) {
+        params.ep_addr = addr_entry->ep_addrs[ep_index].addr;
+        if (uct_ep_is_connected(uct_ep, &params)) {
+            return addr_entry->ep_addrs[ep_index].lane;
+        }
+    }
+
+    return UCP_NULL_LANE;
+}
```

Comment:
it will be needed for next PR of EP reconfiguration, so I preferred not to revert it. (as it would have to be re-reverted for the complete EP reconfiguration)

Reply:
ok